### DISCUSSION TITLE
Custom General Preferences about battery type

### DISCUSSION THREAD

**Discussion by jheredianet** [teslamate-org/teslamate#3513] [General]
Hi, 

I would like to raise a situation that I am seeing lately and that is that the dashboards for cars with LFP batteries are beginning to be duplicated.

![image](https://github.com/teslamate-org/teslamate/assets/45735191/53d48518-dd51-42df-b1c6-977b66ccea2d)

In the end we are going to be filled with dashboards that we are not going to use. I think that what should be done is to have a general preference that allows the user to indicate what type of battery their car has, so that way the panels are displayed according to this preference, in the same way that we have the option to decide the type of unit we need to use (like this, per car in case people use Teslamate for more than one car):

![image](https://github.com/teslamate-org/teslamate/assets/45735191/778dc116-097d-46c6-ba74-7d4057f67d48)

In any case, as the creator of the ["Battery Health" dashboard](https://github.com/jheredianet/Teslamate-CustomGrafanaDashboards), I have seen that it has been cloned in another "Battery Health LFP" and that the changes do not really indicate or have something special about the battery type, but rather the calculations have been modified to consider all charging sessions (short and long), while with time and tests I have done with different car models, I have only considered loads that are greater than the car's default efficiency to avoid dirtying the samples and having a more precise calculation of the degradation.

In my case (Model 3 SR+ 2020) if I use the predefined dashboard, I get 8.5% of estimated degradation:

![image](https://github.com/teslamate-org/teslamate/assets/45735191/3f293e4e-923c-49fe-87f6-d704477368d6)

And with the cloned one I get 8.0%
![image](https://github.com/teslamate-org/teslamate/assets/45735191/a03cdd0c-dbf0-4508-acb4-56f0b0a6b876)

So, for 0.5% of difference, it is not worth having 2 dashboards available. In any case, it would be best to have a dropdown box inside the dashboard itself so that the user can choose the calculations they want to see, and the same for the other dashboards.

In a few days I am going to make some improvements to my main "Battery health" dashboard and I will make the respective Pull Requet and propose that there be a single type of general dashboard for each thing.

Let me know what you think.

**Comment by cwanja** [comment]
See feature request in #3438.

**Reply by jheredianet** [reply]
Thanks, I saw it. From my point of view, it would have been good if we first waited to have the battery type option in Teslamate, before accepting all the PRs for the LFP batteries. I don't see much difference between one type and another so that it is worth having a dashboard for each one

**Reply by cwanja** [reply]
> Thanks, I saw it. From my point of view, it would have been good if we first waited to have the battery type option in Teslamate, before accepting all the PRs for the LFP batteries. I don't see much between one type and another so that it is worth having a dashboard for each one

Completely agree.
